numpy/random/_generator.pyx

```diff
@@ -700,7 +700,7 @@ cdef class Generator:
                 # __index__ must return an integer by python rules.
                 pop_size = operator.index(a.item())
             except TypeError:
-                raise ValueError("a must an array or an integer")
+                raise ValueError("a must be an array or an integer") from None
```

Comment:
Would it make sense to add the type?
```suggestion
                raise ValueError(f"a must be an array or an integer not '{type(a)}'") from None
```

Reply:
Good idea, but at this point, `a` has already been converted to a numpy array, so `type(a)` will always be `<class 'numpy.ndarray'>`.  I pushed a change to avoid that problem.  (I also fixed a couple more error messages that were missing a space.)

Reply:
This is still confusing though, because `random.choice(np.array(1))` will give this message

Reply:
I changed the message to say that `a` must be a sequence or an integer (i.e. replace "array" with "sequence").  Suggestions for better message are welcome!